File: cdb/perl/create/cdb_create.pl
#! /usr/bin/perl    
#
#	cdb_create.pl
#
#					Mar/25/2013
#
# ----------------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
use	Encode;
use	CDB_File;
use	JSON;
#
use lib '/var/www/data_base/common/perl_common';
#
use text_manipulate;
# ----------------------------------------------------------------
print	(encode ('utf-8',"*** 開始 ***\n"));

my $file_cdb = $ARGV[0];

my %dict_aa = data_prepare_proc ();

unlink "$file_cdb" if -e "$file_cdb";
my $cdb = new CDB_File ("$file_cdb", "$file_cdb.$$") or die;

foreach my $key (sort keys %dict_aa){
	my $data = $dict_aa{$key};
	my $json_str = encode_json ($data);
	print ($json_str . "\n");
	$cdb->insert($key,$json_str);
	}

$cdb->finish;

chmod 0666, $file_cdb;

print	(encode ('utf-8',"*** 終了 ***\n"));
#
# ----------------------------------------------------------------
sub data_prepare_proc
{
	my %dict_aa;
	%dict_aa = text_manipulate::dict_append_proc
		('t2761','豊中',84291,'2005-3-22',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2762','池田',27836,'2005-8-5',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2763','高槻',54172,'2005-6-17',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2764','吹田',46832,'2005-8-22',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2765','茨木',27349,'2005-2-28',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2766','摂津',92843,'2005-5-9',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2767','泉大津',38792,'2005-4-10',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2768','和泉',85271,'2005-10-8',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2769','高石',82756,'2005-5-21',%dict_aa);

	return	%dict_aa;
}
# ----------------------------------------------------------------

File: gdbm/php/create/cdb_create.php
#! /usr/bin/php
<?
// ------------------------------------------------------------------
//	cdb_create.php
//
//					Mar/27/2013
//
// ------------------------------------------------------------------
$path="/var/www/data_base/common/php_common";
set_include_path (get_include_path() . PATH_SEPARATOR . $path);

include "text_manipulate.php";
include "dba_manipulate.php";

// ------------------------------------------------------------------
function data_prepare_proc ()
{
$dict_aa = array ();

$dict_aa = dict_append_proc ($dict_aa,'t2761','豊中',51436,'2002-7-24');
$dict_aa = dict_append_proc ($dict_aa,'t2762','池田',47381,'2002-2-8');
$dict_aa = dict_append_proc ($dict_aa,'t2763','高槻',82946,'2002-3-15');
$dict_aa = dict_append_proc ($dict_aa,'t2764','吹田',46321,'2002-10-29');
$dict_aa = dict_append_proc ($dict_aa,'t2765','茨木',35769,'2002-7-14');
$dict_aa = dict_append_proc ($dict_aa,'t2766','摂津',24683,'2002-6-12');
$dict_aa = dict_append_proc ($dict_aa,'t2767','泉大津',82197,'2002-4-17');
$dict_aa = dict_append_proc ($dict_aa,'t2768','和泉',91578,'2002-3-24');
$dict_aa = dict_append_proc ($dict_aa,'t2769','高石',79461,'2002-7-8');

	return	$dict_aa;
}
// ------------------------------------------------------------------

print "*** 開始 ***\n";
$file_cdb=$argv[1];
print $file_cdb . "\n";

$dict_aa = data_prepare_proc ();

$dbh = dba_open($file_cdb,"w","cdb");

foreach ($dict_aa as $key => $value)
	{
	$str_json = json_encode ($value);
	dba_insert($key,$str_json,$dbh);
	}

dba_close ($dbh);
#
print "*** 終了 ***\n";
// ------------------------------------------------------------------
?> 
